src/modules/auth/services/idp.service.ts

```diff
@@ -24,4 +35,40 @@ export class IdpService {
 		})}`;
 		return url;
 	}
+
+	/**
+	 * Flowchart: see https://meemoo.atlassian.net/wiki/spaces/HA2/pages/2978447456/Gebruikersgroepen+en+permissies+BZT+versie+1#Functionele-samenvatting
+	 */
+	public async determineUserGroup(ldapUser: LdapUser): Promise<Group> {
+		// permissions check
+		const apps = get(ldapUser, 'attributes.apps', []);
+		if (
+			!apps.includes('hetarchief') &&
+			!apps.includes('admins') // TODO replace by a single value 'hetarchief-beheer' once archief 2.0 is launched
+		) {
```

Comment:
i would flip this IF check. It's easier to reason about "true" statements than it is to reason about not (!) statements
```
if (
apps.includes('hetarchief') ||
apps.includes('admins') // TODO replace by a single value 'hetarchief-beheer' once archief 2.0 is launched
) {
  // Handle admin logic
}
// Handle visitor/kiosk logic
	
```

Reply:
fixed